### ISSUE TITLE
Crash when selecting or adding a Viewport node to a scene

### ISSUE THREAD

**Issue by noidexe** [godotengine/godot#50792]
### Godot version

7968208b0a5e277a0db7e0d9e265b2eeb3e6b7fe

### System information

Windows 10, GLES 3, GTX 2060

### Issue description

Adding or selecting a viewport node crashes Godot

Backtrace
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] HashMap<StringName,Ref<Font>,HashMapHasherDefault,HashMapComparatorDefault<StringName>,3,8>::getptr (C:\Users\lisan\Documents\GitHub\godot\core\hash_map.h:313)
[1] HashMap<StringName,Ref<Font>,HashMapHasherDefault,HashMapComparatorDefault<StringName>,3,8>::has (C:\Users\lisan\Documents\GitHub\godot\core\hash_map.h:272)
[2] Control::add_font_override (C:\Users\lisan\Documents\GitHub\godot\scene\gui\control.cpp:1886)
[3] TexturePreview::_notification (C:\Users\lisan\Documents\GitHub\godot\editor\plugins\texture_editor_plugin.cpp:56)
[4] TexturePreview::_notificationv (C:\Users\lisan\Documents\GitHub\godot\editor\plugins\texture_editor_plugin.h:39)
[5] Object::notification (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:929)
[6] Control::_notification (C:\Users\lisan\Documents\GitHub\godot\scene\gui\control.cpp:527)
[7] Control::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\control.h:47)
[8] Container::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\container.h:37)
[9] MarginContainer::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\margin_container.h:37)
[10] TexturePreview::_notificationv (C:\Users\lisan\Documents\GitHub\godot\editor\plugins\texture_editor_plugin.h:39)
[11] Object::notification (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:929)
[12] CanvasItem::_enter_canvas (C:\Users\lisan\Documents\GitHub\godot\scene\2d\canvas_item.cpp:540)
[13] CanvasItem::_notification (C:\Users\lisan\Documents\GitHub\godot\scene\2d\canvas_item.cpp:560)
[14] CanvasItem::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\2d\canvas_item.h:162)
[15] Control::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\control.h:47)
[16] Container::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\container.h:37)
[17] MarginContainer::_notificationv (C:\Users\lisan\Documents\GitHub\godot\scene\gui\margin_container.h:37)
[18] TexturePreview::_notificationv (C:\Users\lisan\Documents\GitHub\godot\editor\plugins\texture_editor_plugin.h:39)
[19] Object::notification (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:929)
[20] Node::_propagate_enter_tree (C:\Users\lisan\Documents\GitHub\godot\scene\main\node.cpp:215)
[21] Node::_set_tree (C:\Users\lisan\Documents\GitHub\godot\scene\main\node.cpp:2556)
[22] Node::_add_child_nocheck (C:\Users\lisan\Documents\GitHub\godot\scene\main\node.cpp:1124)
[23] Node::add_child (C:\Users\lisan\Documents\GitHub\godot\scene\main\node.cpp:1141)
[24] EditorInspector::_parse_added_editors (C:\Users\lisan\Documents\GitHub\godot\editor\editor_inspector.cpp:1319)
[25] EditorInspector::update_tree (C:\Users\lisan\Documents\GitHub\godot\editor\editor_inspector.cpp:1453)
[26] EditorInspector::edit (C:\Users\lisan\Documents\GitHub\godot\editor\editor_inspector.cpp:1834)
[27] EditorNode::_edit_current (C:\Users\lisan\Documents\GitHub\godot\editor\editor_node.cpp:2003)
[28] EditorNode::push_item (C:\Users\lisan\Documents\GitHub\godot\editor\editor_node.cpp:1907)
[29] SceneTreeDock::_do_create (C:\Users\lisan\Documents\GitHub\godot\editor\scene_tree_dock.cpp:2161)
[30] SceneTreeDock::_create (C:\Users\lisan\Documents\GitHub\godot\editor\scene_tree_dock.cpp:2198)
[31] MethodBind0<SceneTreeDock>::call (C:\Users\lisan\Documents\GitHub\godot\core\method_bind.gen.inc:149)
[32] Object::call (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:918)
[33] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1226)
[34] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1282)
[35] CreateDialog::_confirmed (C:\Users\lisan\Documents\GitHub\godot\editor\create_dialog.cpp:455)
[36] MethodBind0<CreateDialog>::call (C:\Users\lisan\Documents\GitHub\godot\core\method_bind.gen.inc:149)
[37] Object::call (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:918)
[38] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1226)
[39] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1282)
[40] AcceptDialog::_ok_pressed (C:\Users\lisan\Documents\GitHub\godot\scene\gui\dialogs.cpp:398)
[41] AcceptDialog::_builtin_text_entered (C:\Users\lisan\Documents\GitHub\godot\scene\gui\dialogs.cpp:391)
[42] MethodBind1<AcceptDialog,String const &>::call (C:\Users\lisan\Documents\GitHub\godot\core\method_bind.gen.inc:867)
[43] Object::call (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:918)
[44] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1226)
[45] Object::emit_signal (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:1282)
[46] LineEdit::_gui_input (C:\Users\lisan\Documents\GitHub\godot\scene\gui\line_edit.cpp:323)
[47] MethodBind1<LineEdit,Ref<InputEvent> >::call (C:\Users\lisan\Documents\GitHub\godot\core\method_bind.gen.inc:867)
[48] Object::call_multilevel (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:766)
[49] Object::call_multilevel (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:864)
[50] Viewport::_gui_input_event (C:\Users\lisan\Documents\GitHub\godot\scene\main\viewport.cpp:2374)
[51] Viewport::input (C:\Users\lisan\Documents\GitHub\godot\scene\main\viewport.cpp:2786)
[52] Viewport::_vp_input (C:\Users\lisan\Documents\GitHub\godot\scene\main\viewport.cpp:1401)
[53] MethodBind1<Viewport,Ref<InputEvent> const &>::call (C:\Users\lisan\Documents\GitHub\godot\core\method_bind.gen.inc:867)
[54] Object::call (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:918)
[55] Object::call (C:\Users\lisan\Documents\GitHub\godot\core\object.cpp:847)
[56] SceneTree::call_group_flags (C:\Users\lisan\Documents\GitHub\godot\scene\main\scene_tree.cpp:270)
[57] SceneTree::input_event (C:\Users\lisan\Documents\GitHub\godot\scene\main\scene_tree.cpp:429)
[58] InputDefault::_parse_input_event_impl (C:\Users\lisan\Documents\GitHub\godot\main\input_default.cpp:435)
[59] InputDefault::parse_input_event (C:\Users\lisan\Documents\GitHub\godot\main\input_default.cpp:260)
[60] InputDefault::flush_accumulated_events (C:\Users\lisan\Documents\GitHub\godot\main\input_default.cpp:653)
[61] OS_Windows::process_events (C:\Users\lisan\Documents\GitHub\godot\platform\windows\os_windows.cpp:2534)
[62] OS_Windows::run (C:\Users\lisan\Documents\GitHub\godot\platform\windows\os_windows.cpp:3307)
[63] widechar_main (C:\Users\lisan\Documents\GitHub\godot\platform\windows\godot_windows.cpp:161)
[64] _main (C:\Users\lisan\Documents\GitHub\godot\platform\windows\godot_windows.cpp:183)
[65] main (C:\Users\lisan\Documents\GitHub\godot\platform\windows\godot_windows.cpp:195)
[66] __scrt_common_main_seh (d:\agent\_work\4\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[67] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### Steps to reproduce

- Create a new project
- Create a new scene with any root node
- Add a viewport node
Expected: 
- The node is added and is editable
Actual:
- The editor crashes

**Comment by akien-mga** [comment]
To be clear, are you sure that you're on the referenced commit? Because this issue (or a related one) should have been fixed today in `master`.

**Comment by noidexe** [comment]
I'm on 3.x and built from that commit. I guess the fix hasn't been backported yet then.

**Comment by akien-mga** [comment]
Ah right, there's this hotfix still need for 3.x: https://github.com/godotengine/godot/pull/50777

**Comment by noidexe** [comment]
I just cherry picked the commit and it know works. Thanks!

**Comment by akien-mga** [comment]
Fixed by 49f323410bfc2c2df15b8f4321c79da7e402ee1e.
